lib/collection/src/collection/shard_transfer.rs

```diff
@@ -140,6 +140,10 @@ impl Collection {
         let shard_holder = self.shards_holder.clone();
         let collection_id = self.id.clone();
         let channel_service = self.channel_service.clone();
+        let default_shard_transfer_method = self
+            .shared_storage_config
+            .default_shard_transfer_method
+            .unwrap_or_default();
```

Comment:
What if you specify `wal_delta` as default shard transfer method in the config file (I believe it's `storage.shard_transfer_method` config parameter)? 🤔

I might be wrong here, but I also believe we are using the same parameter, *[when requesting shard transfers during `Collection::sync_local_state` in the first place](https://github.com/qdrant/qdrant/blob/dev/lib/collection/src/collection/mod.rs#L573-L578)*.

So:
- if you specify `shard_transfer_method: wal_delta`, then our `sync = true` transfers would use WAL diff by default... but then if WAL diff transfer fails, it will try to fallback to WAL diff transfer once again
- if you specify `shard_transfer_method: stream_records`, then WAL diff would properly fallback to stream records transfer... but then `sync = true` transfers would always use stream records, and you could only start WAL diff transfer manually with the API

I think we need to introduce *separate* parameter for WAL diff fallback method. And check that it's *not* `wal_delta` when deserializing config. 😂

Also, I think it's a bit confusing that it's `StorageConfig::shard_transfer_method`, but `SharedStorageConfig::default_shard_transfer_method`.

Reply:
I also see how feature gating is implemented in #3800 now.

@generall mentioned this week, that he wants a config parameter to be able to disable WAL diff transfers for `sync = true` transfers (even if supported; like, as an emergency measure, if WAL diffs happen to work poorly in practice), so I think we still want/need to introduce a few parameters, but it will take a bit more thinking how to do so properly.

Reply:
> What if you specify wal_delta as default shard transfer method in the config file (I believe it's storage.shard_transfer_method config parameter)? 🤔

> I might be wrong here, but I also believe we are using the same parameter, [when requesting shard transfers during Collection::sync_local_state in the first place](https://github.com/qdrant/qdrant/blob/dev/lib/collection/src/collection/mod.rs#L573-L578).

The same property is used, correct. But if it tries to fall back to WAL delta transfer again, it simply does not fall back at all. So there is no loop here.

Anyway. @generall mentioned that we want to fall back to streaming records in all cases for now, because snapshot transfer is not super robust yet. In that case we don't need a property to configure it, and can add this later if we so desire if snapshot transfers are up to speed again. I agree with that idea.

Reply:
I've now implemented this. We fall back to stream records, no matter what is configured in the configuration.